src/host/consoleInformation.cpp

```diff
@@ -44,42 +46,41 @@ CONSOLE_INFORMATION::CONSOLE_INFORMATION() :
 {
     ZeroMemory((void*)&CPInfo, sizeof(CPInfo));
     ZeroMemory((void*)&OutputCPInfo, sizeof(OutputCPInfo));
-    InitializeCriticalSection(&_csConsoleLock);
 }
 
-CONSOLE_INFORMATION::~CONSOLE_INFORMATION()
-{
-    DeleteCriticalSection(&_csConsoleLock);
-}
+static thread_local ULONG recursionCount = 0;

```

Comment:
I initially tried to fix our code to not depend on recursive locking, but noticed that it's not possible to fix. Because the WndProc can be called recursively by Windows, we're forced to make our locks reentrant for conhost as well.

Reply:
Gross. OK.